### ISSUE TITLE
Make ip_address optional for digitalocean_domain (as it is in DigitalOcean’s API)

### ISSUE THREAD

**Issue by vandrijevik** [terraform-providers/terraform-provider-digitalocean#112]
### Terraform Version
```
$ terraform -v
Terraform v0.11.7
+ provider.digitalocean v0.1.3
```

### Affected Resource(s)
- `digitalocean_domain`

### Expected Behavior
In [Terraform documentation](https://www.terraform.io/docs/providers/do/r/domain.html), `ip_address` is listed as a required attribute, with the justification:

> (Required) The IP address of the domain. This IP is used to created an initial A record for the domain. It is required upstream by the DigitalOcean API.

However, [the attribute `ip_address` is optional in the DigitalOcean API](https://developers.digitalocean.com/documentation/v2/#create-a-new-domain), and API calls without it successfully create a domain without an initial A record.

Terraform’s DigitalOcean provider should allow creating `digitalocean_domain` resources without specifying `ip_address`. 

### Actual Behavior
Terraform’s DigitalOcean provider requires specifying an `ip_address` for `digitalocean_domain` resources, forcing the creation of an initial A record for the apex domain.

**Comment by Temikus** [comment]
It's causing issues with import too. Importing a domain doesn't actually get the A record:
```
$ terraform import digitalocean_domain.example "example.com"
```
So the resource gets imported in a state that cannot be managed (forces recreation):
```
$ terraform show
digitalocean_domain.example:
  id = example.com
  name = example.com
```
```
$ cat domain.tf

resource "digitalocean_domain" "example" {
  name          = "example.com"
  ip_address = "8.8.8.8" #Already set to 8.8.8.8 in DO
}


$ terraform plan
...
-/+ digitalocean_domain.example (new resource required)
      id:                                                  "example.com" => <computed> (forces new resource)
      ip_address:                                          "" => "8.8.8.8" (forces new resource)
      name:                                                "example.com" => "example.com"
...
```

**Comment by sagikazarmark** [comment]
I started working on a fix, but digitalocean/godo#172 is a blocker.

**Comment by andrewsomething** [comment]
@sagikazarmark godo made a new release last week with that change included. So this should now be unblocked. Let me know if you're still interested in working on this change. Thanks!

**Comment by sagikazarmark** [comment]
Thanks @andrewsomething I missed that. I've just submitted a PR, but closed it in favor of #122

**Comment by sagikazarmark** [comment]
🎉
